airflow/executors/kubernetes_executor.py

```diff
@@ -863,9 +876,12 @@ def _flush_result_queue(self) -> None:
 
     def end(self) -> None:
         """Called when the executor shuts down"""
-        assert self.task_queue, NOT_STARTED_MESSAGE
-        assert self.result_queue, NOT_STARTED_MESSAGE
-        assert self.kube_scheduler, NOT_STARTED_MESSAGE
+        if not self.task_queue:
```

Comment:
```suggestion
        if not (self.task_queue and self.queue_result and self.kube_scheduler):
```

Reply:
Same here.  I specifically want to keep those checks separately - so that you can figure out from the stack trace which of them was None. This is much easier than to do debugging.

Reply:
Same here @nuclearpinguin ?